daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -498,7 +496,7 @@ private final class `Value Order instance`[Cid: Order](Scope: Value.LookupVarian
           ctorOrder(idA, idB, a, b)
       })
     case ValueRecord(_, a) => (140, k { case ValueRecord(_, b) => _2.T.subst(a) ?|? _2.T.subst(b) })
-    case ValueStruct(a) => (150, k { case ValueStruct(b) => a ?|? b })
+    case ValueStruct(a) => (150, k { case ValueStruct(b) => a.toImmArray ?|? b.toImmArray })
```

Comment:
```suggestion
    case ValueStruct(a) => (150, k { case ValueStruct(b) => a ?|? b })
```

`Struct` should have an `Order` and `Equal` instance modeled like that for `ImmArray`.

Reply:
done